Feature: add function for checking misbehaviors in Vetomint

Jeongseup opened the pull request.

Jeongseup commented:
force push means for removing noise comments

junha1 reviewed (commented):
I think the overall structure should be revised.
1. Reporting misbehaviors should be done on the same basis of feeding ConsensusEvents. However, the current interface of `check_misbehavior()` is confusing. Why does it take `check_round` and `check_proposal`? When is the function supposed to be called? Why did you take those two very particular parameters?
2. Don't use `println!()`.It seems you just printed whenever there is no misbehavior. This is just so random.
3. Please add some comments about important code. For example, I'd put doc comments for every `check_~` functions.
4. The local variable `valid_prevotes` is named very misleadingly. Why is it `valid`? You just filter prevotes by their rounds.

Jeongseup commented:
I modified with your review. but, I don't fully understand the vetomint. so I'm not sure which place i called the check~ function is correct. and some check_invalid_~ functions is in unimplement.

ans1) I deleted the `check_misbehavior` function. because it didn't seem necessary when I looked at it again. 
Instead of, I add some check misbehavior functions for responsing `ConsensusResponse::ViolationReport` enum which will be used in (https://github.com/postech-dao/simperby/blob/main/consensus/src/state.rs#L225C7-L225C7)

ans2) I removed all `println!()` macro in my PR

ans3) I already know what comment I should put for evenry `check~` function. but I'm not sure what to say. so, Can you give me some guideline for me?

ans4) I changed `unimplement` state at some functions which had `valid_prevotes` local variables. 

---

and
